### ISSUE TITLE
Regression on Style/MultipleComparison: false positives

### ISSUE THREAD

**Issue by akimd** [rubocop/rubocop#13193]
In the following example, there are several "multiple comparisons", but each one of them has only two comparisons, not more.  I have:
```
Style/MultipleComparison:
  ComparisonsThreshold: 3
```

so they are all right.  However there must be some state kept somewhere and the cop believes we reached the threshold.

```ruby
$ cat multiple-comparison.rb
def foo(stage)
  puts if stage == 1 || stage == 2
end

def bar(safe)
  puts if safe == 'false' || safe == 'true'
end

def baz(x, y)
  puts if x == 1 || x == 2
  puts if y == 1 || y == 2
end
```

--------

## Expected behavior

No complaints.   That was the case with 1.65.1, it's a regression in 1.66 and 1.66.1.

## Actual behavior

```
$ bundle exec rubocop --only Style/MultipleComparison multiple-comparison.rb
Inspecting 1 file
C

Offenses:

multiple-comparison.rb:6:11: C: [Correctable] Style/MultipleComparison: Avoid comparing a variable with multiple items in a conditional, use Array#include? instead.
  puts if safe == 'false' || safe == 'true'
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
multiple-comparison.rb:11:11: C: [Correctable] Style/MultipleComparison: Avoid comparing a variable with multiple items in a conditional, use Array#include? instead.
  puts if y == 1 || y == 2
          ^^^^^^^^^^^^^^^^

1 file inspected, 2 offenses detected, 2 offenses autocorrectable
```

If I remove `foo`, then `bar` has no violation.

## RuboCop version

```
$ bundle exec rubocop -V
1.66.1 (using Parser 3.3.5.0, rubocop-ast 1.32.2, running on ruby 3.3.4) [arm64-darwin23]
  - rubocop-performance 1.21.1
```

**Comment by Earlopain** [comment]
@fatkodima, do you want to take a look?

**Comment by fatkodima** [comment]
Yes, I will take a look.

**Comment by vlad-pisanov** [comment]
@fatkodima I took a crack at [fixing the regression](https://github.com/rubocop/rubocop/pull/13195), can you take a look when you get a chance? 🙏
